### PR TITLE
CH-679-CH-738 put the password update into a new endpoint

### PR DESCRIPTION
**Pull request by NatalliaBulyha**
jira[endpoint: change password in personal account] https://cozyhometeam.atlassian.net/browse/CH-738?atlOrigin=eyJpIjoiNWI0NTU5MDczMjYyNDA3MzhjYmM3MzUxYmNmYzY2N2IiLCJwIjoiaiJ9

### CODE REVIEW

File: src/main/java/com/cozyhome/onlineshop/userservice/controller/UserSecuredController.java [hunk 1]
```diff
@@ -63,7 +64,19 @@ public ResponseEntity<UserInformationResponse> updateUserInfo(@RequestBody @Vali
         return ResponseEntity.ok(userService.updateUserData(userInformationDto, userId));
     }
 
-    @Operation(summary = "Get user information")
+    @Operation(summary = "Update user password in personal account")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = SwaggerResponse.Code.CODE_200, description = SwaggerResponse.Message.CODE_200_FOUND_DESCRIPTION) })
+    @Secured({"ROLE_CUSTOMER"})
```
Comment (DariaBoo) [inline src/main/java/com/cozyhome/onlineshop/userservice/controller/UserSecuredController.java]: propose to add "ROLE_MANAGER", "ROLE_ADMIN", admin and manager can also have personal account and should have ability to change personal data

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -27,163 +20,174 @@
...
-		user.setModifiedAt(LocalDateTime.now());
-		user.setStatus(UserStatusE.DELETED);
-		log.info("[ON deleteUser] :: changed the user status to DELETED for the user with email {}", email);
-		userRepository.save(user);
-	}
+    }
+
+    @Override
+    public void updateUserPassword(PasswordUpdateRequest passwords, String userId) {
+        User user = userRepository.findByIdAndStatus("6525835b1578af3164f1c4d3", UserStatusE.ACTIVE)
```
Comment (DariaBoo) [inline src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: "6525835b1578af3164f1c4d3" change to userId

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -27,163 +20,174 @@
...
+                .orElseThrow(() -> new DataNotFoundException("Unable to update password. User not found."));
+        user.setPassword(encoder.encode("admiN1$2"));
```
Comment (DariaBoo) [inline src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: "admiN1$2" change to passwords.getNewPassword()

Comment (DariaBoo) [reply src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: suppose it was a test code

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -27,163 +20,174 @@
...
+    public UserInformationResponse getUserInfo(String userId) {
+        User user = userRepository.findByIdAndStatus(userId, UserStatusE.ACTIVE)
+                .orElseThrow(() -> new DataNotFoundException(String.format("User with id = %s not found.", userId)));
+        return userBuilder.buildUserInformationResponse(user);
+    }
+
+    @Override
+    public void deleteUser(String email) {
+        User user = userRepository.findByEmailAndStatus(email, UserStatusE.ACTIVE)
+                .orElseThrow(() -> new IllegalArgumentException("Not user found by the email " + email));
```
Comment (DariaBoo) [inline src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: propose to use the same exceptions: IllegalArgumentException or DataNotFoundException. In this class were used both.

File: src/main/java/com/cozyhome/onlineshop/userservice/controller/UserSecuredController.java (hunk 1 above)
Comment (NatalliaBulyha) [reply src/main/java/com/cozyhome/onlineshop/userservice/controller/UserSecuredController.java]: done

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (NatalliaBulyha) [reply src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: done

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (NatalliaBulyha) [reply src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: done

File: src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java (hunk 4 above)
Comment (NatalliaBulyha) [reply src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java]: it's a bug in our logic, so I use custom exceptions. I will use DataNotFoundException

**Pull request merged**